src/ensembl/src/content/app/custom-download/components/checkbox-with-selects/CheckboxWithSelects.tsx

```diff
@@ -7,25 +7,32 @@ import ImageButton from 'src/shared/image-button/ImageButton';
 import { ReactComponent as AddIcon } from 'static/img/browser/zoom-in.svg';
 import { ReactComponent as RemoveIcon } from 'static/img/shared/clear.svg';
 
-type Props = {
+export type CheckboxWithSelectsProps = {
   options: Option[];
   label: string;
   selectedOptions: string[];
+  disabled?: boolean;
   onChange: (selectedOptions: string[]) => void;
 };
 
-const CheckboxWithSelects = (props: Props) => {
-  const [isChecked, setisChecked] = useState(props.selectedOptions.length > 0);
+const CheckboxWithSelects = (props: CheckboxWithSelectsProps) => {
+  const [isChecked, setIsChecked] = useState(false);
   const [shouldShowExtraOption, setShowExtraOption] = useState(false);
 
+  useEffect(() => {
+    setIsChecked(props.selectedOptions.length > 0);
+  }, [props.selectedOptions]);
+
   const handleCheckboxOnChange = (isChecked: boolean) => {
-    setisChecked(isChecked);
-    props.onChange([]);
+    setIsChecked(isChecked);
+    if (!isChecked && props.selectedOptions.length > 0) {
+      props.onChange([]);
+    }
   };
 
   const handleOnSelect = (value: string, selectIndex?: number) => {
     const newSelectedOptions: string[] = [...props.selectedOptions];
-    if (selectIndex) {
+    if (selectIndex !== undefined) {
       newSelectedOptions[selectIndex] = value;
     } else {
       newSelectedOptions.push(value);
```

Comment:
line 48 (can’t leave comment directly): ` if (selectedOptionsClone.indexOf(option.value) === -1) ` ->  `if (selectedOptionsClone.includes(option.value))`

Reply:
also, `removeSelection` looks like it could be written with a simple `filter` function

Reply:
That's done